src/index.ts

```diff
@@ -336,6 +337,44 @@ function suppressNoSubscriptionExistsWarning<T>(f: () => T): T {
   }
 }
 
+/**
+ * Another VERY hacky monkeypatch to make sure that we can take extra parameters to certain alchemy subscriptions
+ * I hate doing this, but the other option is to fork web3-core and I think for now this is better. Any web3-core
+ * update to this method will have to be manually duplicated by us here.
+ */
+const web3CoreSubscriptions = require("web3-core-subscriptions"); // tslint:disable-line:no-var-requires
+const web3CoreErrors = require("web3-core-helpers").errors; // tslint:disable-line:no-var-requires
+web3CoreSubscriptions.subscription.prototype._validateArgs = function (
+  args: any,
+) {
+  let subscription: any = this.options.subscription;
+  if (!subscription) {
+    subscription = {};
+  }
```

Comment:
Can be written more simply as
```ts
const subscription = this.options.subscription ?? {};
```
or
```ts
const { subscription = {} } = this.options;
```

Reply:
Oh I see this was copied verbatim from the original web3 source, so makes sense. Regardless if you go with the suggestion to wrap the old method above, then this won't be necessary.